### PR TITLE
fixes the array flattening option

### PR DESCRIPTION
**Pull request by ovidiul**
## Description
This PR fixes an issue with displaying the aggregated filters widget by properly flattening an array tree.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

### CODE REVIEW

File: includes/utils.php [hunk 1]
```diff
@@ -416,27 +416,32 @@ function get_term_tree( $all_terms, $orderby = 'count', $order = 'desc', $flat =
...
+    $result = [];
+    foreach( $arr as $item ) {
+        if ( isset($item->children ) ) {
+            $result = array_merge( $result, flatten_tree_array( $item->children ) );
+		}
+        unset( $item->children );
+        $result[] = $item;  
+    }
+    return $result;
+}
```
Comment (pschoffer) [inline includes/utils.php]: Fixing indentation 

```suggestion
function flatten_tree_array( $arr ) {
	$result = [];
	foreach ( $arr as $item ) {
		if ( isset( $item->children ) ) {
			$result = array_merge( $result, flatten_tree_array( $item->children ) );
		}
		unset( $item->children );
		$result[] = $item;  
	}
	return $result;
}
```

**Review by pschoffer — CHANGES_REQUESTED** [review]

File: includes/utils.php (hunk 1 above)
Comment (ovidiul) [reply includes/utils.php]: Indentation fixed now

**Comment by ovidiul** [comment]
Hi @pschoffer indentation is fixed now, I have also moved the tests from #88 PR here and refactored them to include the suggested data provider, thanks for the suggestion.

File: includes/utils.php [hunk 2]
```diff
@@ -416,27 +416,35 @@ function get_term_tree( $all_terms, $orderby = 'count', $order = 'desc', $flat =
...
-				$flat_tree[] = $term;
-
-				if ( ! empty( $term->children ) ) {
-					$to_process = $term->children + $to_process;
-				}
-			}
-		}
+		$flat_tree =  flatten_tree_array( $terms_tree );
 
 		return $flat_tree;
```
Comment (pschoffer) [inline includes/utils.php]: Please replace with the upstream fix:
https://github.com/10up/ElasticPress/commit/cd4183708c63f45a064220eb4b19866d1f417f37

So changing `$to_process = $term->children + $to_process;` to `$to_process = array_merge( $term->children, $to_process );`

The reason this was failing for your use case. is that `+` will not merge items with the same index.

**Review by pschoffer — CHANGES_REQUESTED** [review]

**Review by pschoffer — APPROVED** [review]
Perfect, thank you for this!

**Pull request merged**